.pre-commit-config.yaml

```diff
@@ -8,90 +8,109 @@ repos:
+    exclude: plasmapy/tests/_helpers/.*
+  - id: trailing-whitespace
+  - id: end-of-file-fixer
   - id: check-json
   - id: check-toml
   - id: check-yaml
 
-- repo: https://github.com/tox-dev/pyproject-fmt
-  rev: 1.7.0
+- repo: https://github.com/python-jsonschema/check-jsonschema
+  rev: 0.27.4
   hooks:
-  - id: pyproject-fmt
+  - id: check-github-workflows
 
-- repo: https://github.com/astral-sh/ruff-pre-commit
-  rev: v0.1.15
+- repo: https://github.com/sirosen/texthooks
+  rev: 0.6.4
   hooks:
-  - id: ruff
-    args: [--fix]
-
-- repo: https://github.com/PyCQA/isort
-  rev: 5.13.2
-  hooks:
-  - id: isort
-    name: isort
-    entry: isort
-    require_serial: true
-    language: python
-    types:
-    - python
+  - id: fix-smartquotes
+  - id: fix-spaces
+  - id: fix-ligatures
+  - id: forbid-bidi-controls
 
-- repo: https://github.com/psf/black
-  rev: 24.1.1
+- repo: https://github.com/Lucas-C/pre-commit-hooks
+  rev: v1.5.4
   hooks:
-  - id: black
-
-- repo: https://github.com/nbQA-dev/nbQA
-  rev: 1.7.1
-  hooks:
-  - id: nbqa-black
-    additional_dependencies:
-    - black==24.1.1
-    args:
-    - --nbqa-mutate
-  - id: nbqa-isort
-    additional_dependencies:
-    - isort==5.12.0
-    args:
-    - --nbqa-mutate
-
+  - id: remove-crlf
+    name: remove CRLF line endings
 - repo: https://github.com/macisamuele/language-formatters-pre-commit-hooks
   rev: v2.12.0
   hooks:
   - id: pretty-format-ini
     args: [--autofix]
   - id: pretty-format-yaml
     args: [--autofix]
+    # For the labeler GitHub Action, labels with spaces in them must
+    # be put in quotes. However, the pretty-format-yaml hook will
+    # remove the quotes which will break that action (and certain other
+    # actions).
+    exclude: .github/labeler.yml|.pre-commit-search-and-replace.yaml
 
 - repo: https://github.com/MarcoGorelli/absolufy-imports
   rev: v0.3.1
   hooks:
   - id: absolufy-imports
-    exclude: docs/plasmapy_sphinx
+    name: Use absolute rather than relative imports
+
+- repo: https://github.com/pre-commit/pygrep-hooks
+  rev: v1.10.0
+  hooks:
+  - id: python-check-blanket-noqa
+    name: noqa comments have an error code
+    exclude: ^docs/plasmapy_sphinx/
+  - id: rst-directive-colons
+  - id: rst-inline-touching-normal
+  - id: text-unicode-replacement-char
+
 
 - repo: https://github.com/codespell-project/codespell
   rev: v2.2.6
   hooks:
   - id: codespell
-    exclude: .*\.fits
+    name: codespell (add false positives to pyproject.toml)
+    args: [--write-changes]
     additional_dependencies:
     - tomli
+    exclude: .*\.fits
 
-- repo: https://github.com/asottile/pyupgrade
-  rev: v3.15.0
+- repo: https://github.com/astral-sh/ruff-pre-commit
+  rev: v0.1.15
   hooks:
-  - id: pyupgrade
-    args: [--keep-runtime-typing, --py39-plus]
-    files: ^xrtpy/
+  - id: ruff
+    name: ruff (see https://docs.astral.sh/ruff/rules)
+    args: [--fix]
+  - id: ruff-format
+    name: autoformat source code with ruff formatter
```

Comment:
We can use `ruff-format` to replace `black` as our autoformatter.  It works pretty similarly to `black`, but is significantly faster since `ruff` tools are written in the Rust language.